app/models/push.rb

```diff
@@ -2,31 +2,42 @@
 
 class Push < ActiveRecord::Base
   fields do
-    status :string, limit: 32
-    email_sent :boolean, default: false
+    status       :string,  limit: 32
+    email_sent   :boolean, default: false
+
     timestamps
   end
 
   validates :status, inclusion: Github::Api::Status::STATES.map(&:to_s)
 
-  belongs_to :head_commit, class_name: 'Commit', required: true
   has_many :commits_and_pushes, class_name: :CommitsAndPushes, inverse_of: :push, dependent: :destroy
   has_many :commits, through: :commits_and_pushes
   has_many :jira_issues_and_pushes, class_name: :JiraIssuesAndPushes, inverse_of: :push, dependent: :destroy
   has_many :jira_issues, through: :jira_issues_and_pushes
+
+  belongs_to :head_commit, class_name: 'Commit', required: true
   belongs_to :branch, inverse_of: :pushes, required: true
+  belongs_to :ancestor_ref, class_name: 'AncestorRef', required: true
```

Comment:
Similar to @caitlinrhd's question about creating these manually and backfilling, making this a required field could be problematic without a backfill script (could be done via SQL as a manually written migration file) because without data in the column for a record it's possible that the records are then considered invalid and that could cause problems elsewhere

Reply:
Great point! I added a `Push.update_all` in the migration file. I haven't had to do this before, so let me know if there's a better way to do this https://github.com/Invoca/pre_deploy_checker/blob/e21192e40aea979223efa879b2d58b41611ca445/db/migrate/20200922230337_create_services.rb#L14-L21